### PR TITLE
[Refac] Use native pytorch higher-order blocks and Adjoint Differentiation

### PR DESCRIPTION
**Pull request by dominikandreasseitz**
pyqtorch recently introduced several higher order operations like `Sequence`, `Add` and `Scale` which can now be used via qadence

### CODE REVIEW

File: qadence/backends/adjoint.py [hunk 1]
```diff
@@ -105,27 +106,28 @@ def _apply_adjoint(ctx: Any, op: Module) -> list:
                 ctx.projected_state = apply_operator(
                     ctx.projected_state, op.dagger(values), op.qubit_support
                 )
-            elif isinstance(op, ScalePyQOperation):
-                ctx.out_state = apply_operator(ctx.out_state, op.dagger(values), op.qubit_support)
+            elif isinstance(op, pyq.Scale):
```
Comment (Roland-djee) [inline qadence/backends/adjoint.py:109]: Couldn't you simply import them ? I guess there is no ambiguity.

**Review by Roland-djee — APPROVED** [review]
Thanks @dominikandreasseitz looks nice and clean.

**Review by jpmoutinho — APPROVED** [review]
Nice :)

File: qadence/backends/adjoint.py (hunk 1 above)
Comment (dominikandreasseitz) [reply qadence/backends/adjoint.py]: file removed

**Pull request merged**